torch/utils/data/distributed.py

```diff
@@ -65,3 +65,116 @@ def __len__(self):
+    is a separate process with an identical copy of sampler.
+    Because of that, on a distributed environment, different processes
+    could read the same portion of the dataset. This sampler uses strides,
+    based on the global worker rank concept, to coordinate parallel reading of chunks of data.
+
+    For example, assume 2 workers reading the same `ChunkDataset` dataset.
+    Each worker process needs to configure their `DistributedChunkSampler`
+    so that one of them reads all even batches (e.g. `rank` 0)
+    while the other process reads all odd batches (e.g. `rank` 1).
+
+    If :attr:`shuffle` is ``True``, sampler will shuffle the indicies
+
+    Args:
+        num_replicas (int): Number of processes participating in the data reading.
+            Must be equal to the number of distributed processes * number of DataLoader workers.
+        rank (int, optional): Current rank of the worker
+            Internally, :attr:`rank` will be recalculated on each `DataLoader` worker
+            based on current :attr:`rank`, and `DataLoader` information
+            such as number of workers and worker ID, following the expression:
+                `rank`=`worker_id`+`current_rank`*`num_workers`. (default: ``0``)
+        num_chunks (int, optional): Number of chunks participating in the sampling.
+            (default: ``0``)
+        shuffle (bool, optional): set to ``True`` to have the chunk indices reshuffled.
+            (default: ``False``)
+    """
+
+    def __init__(self, num_replicas, rank=0, num_chunks=0, shuffle=False):
+        super(DistributedChunkSampler, self).__init__(None)
+        assert rank >= 0, 'rank must be >= 0'
+        assert num_replicas >= 0, 'num_replicas must be >= 0'
+        assert rank <= num_replicas, 'rank must be <= num_replicas'
+        assert num_chunks >= num_replicas, 'num_chunks must be >= num_replicas'
+        self.rank = rank
+        self.num_replicas = num_replicas
+        self.num_chunks = num_chunks
+        self.shuffle = shuffle
+        self.epoch = 0
+
+    def __iter__(self):
+        r"""Stride logic is implemented here"""
+
+        # Deterministically shuffle based on epoch
+        g = torch.Generator()
+        g.manual_seed(self.epoch)
+        if self.shuffle:
+            indices = torch.randperm(self.num_chunks, generator=g).tolist()
+        else:
+            indices = list(range(self.num_replicas))
+
+        # Create a strided sublist
+        indices = indices[self.rank:self.num_chunks:self.num_replicas]
+        return iter(indices)
+
+    def __len__(self):
+        raise NotImplementedError
+
+    def set_rank(self, rank):
+        r"""Sets current global worker rank
+
+        Typically used after new processes are spawned with a copy of this sampler
+        to prevent sampling the same indices in different workers
+        """
+        assert rank >= 0, 'rank must be >= 0'
+        assert rank < self.num_replicas, 'rank must < num_replicas'
```

Comment:
here, the num_replica needs to be world_size * num_workers. Did we make this clear?

Reply:
At `DistributedChunkSampler` documentation (distributed.py:126), in the "Argument" section I have added the following:

> Args:
>         num_replicas (int): Number of processes participating in the data reading.
>             Must be equal to the number of distributed processes * number of DataLoader workers.